scripts/scraper/process-macros/extract-macro-calls.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * Given the name of a macro and some HTML,
+ * find the all invocations of the macro in the HTML.
+ *
+ * Return an array, containing one element per invocation,
+ * where each element is itself an array containing the arguments
+ * with which the macro was called in that invocation.
+ */
+function extractMacroCalls(macroName, html) {
+  const regex = new RegExp(`{{\\ ?${macroName}\\((.*?)\\) ?}}`, 'gi');
+  let allInvocations = [];
+  let thisInvocation = [];
+
+  while ((thisInvocation = regex.exec(html)) !== null) {
```

Comment:
This is sort of cryptic. I had to re-read how `exec` works to understand that this loop will eventually exhaust the matches. I don't have any idea how to make this less cryptic, so I'd appreciate a comment here to give a hint about what's going on

Reply:
Yes, I agree this is cryptic. I just copied the code out of MDN :).